### ISSUE TITLE
Tabs cause a emitter to reprint the same line repeatedly

### ISSUE THREAD

**Issue by mr-cal** [canonical/craft-cli#182]
### Bug Description

If there is a tab in an emitter message, it causes the line to be reprinted every time the spinner spins.

### To Reproduce

```python
from craft_cli import EmitterMode, emit
from pathlib import Path
from time import sleep

emit.init(
    mode=EmitterMode.TRACE,
    appname="test",
    greeting="hello",
    log_filepath=Path("log.txt"),
    streaming_brief=True,
)

emit.progress("\tmessage")
sleep(5)
```

### Expected output
```
2023-10-04 11:30:18.460 hello
2023-10-04 11:30:18.460 Logging execution to 'log.txt'
2023-10-04 11:30:18.460     message \ (4.9s)
```

### Actual output
```
2023-10-04 11:28:58.446 hello
2023-10-04 11:28:58.446 Logging execution to 'log.txt'
2023-10-04 11:28:58.446 	message2023-10-04 11:28:58.446 	message - (2.0s)2023-10-04 11:28:58.446 	message \ (2.1s)2023-10-04 11:28:58.446 	message | (2.2s)2023-10-04 11:28:58.446 	message / (2.3s)2023-10-04 11:28:58.446 	message - (2.4s)2023-10-04 11:28:58.446 	message \ (2.5s)2023-10-04 11:28:58.446 	message | (2.6s)2023-10-04 11:28:58.446 	message / (2.7s)2023-10-04 11:28:58.446 	message - (2.8s)2023-10-04 11:28:58.446 	message \ (2.9s)2023-10-04 11:28:58.446 	message | (3.0s)2023-10-04 11:28:58.446 	message / (3.1s)2023-10-04 11:28:58.446 	message - (3.2s)2023-10-04 11:28:58.446 	message \ (3.3s)2023-10-04 11:28:58.446 	message | (3.4s)2023-10-04 11:28:58.446 	message / (3.5s)2023-10-04 11:28:58.446 	message - (3.6s)2023-10-04 11:28:58.446 	message \ (3.7s)2023-10-04 11:28:58.446 	message | (3.8s)2023-10-04 11:28:58.446 	message / (3.9s)2023-10-04 11:28:58.446 	message - (4.0s)2023-10-04 11:28:58.446 	message \ (4.1s)2023-10-04 11:28:58.446 	message | (4.2s)2023-10-04 11:28:58.446 	message / (4.3s)2023-10-04 11:28:58.446 	message - (4.4s)2023-10-04 11:28:58.446 	message \ (4.5s)2023-10-04 11:28:58.446 	message | (4.6s)2023-10-04 11:28:58.446 	message / (4.7s)2023-10-04 11:28:58.446 	message - (4.8s)2023-10-04 11:28:58.446 	message \ (4.9s)
```

The above displays normally (see picture attached) but it copy/pastes like there are no newlines.
![Screenshot_20231004_113327](https://github.com/canonical/craft-cli/assets/60674096/14697ade-def7-4b44-9e86-ce3651d0e15b)

**Comment by tigarmo** [comment]
Thanks for reporting! Out of curiosity, what is the size of the tab in the
environment where you are running this? This issue rings a bell

**Comment by mr-cal** [comment]
> Thanks for reporting! Out of curiosity, what is the size of the tab in the environment where you are running this? This issue rings a bell

A tab evaluates to 8 spaces in my bash environment.

**Comment by lengau** [comment]
@mr-cal can you verify that this is fixed on main? (It's not yet released, we'll need 2.8.0 for it I think)
